adaptors/dummy/deploy/e2e/usability/usability_test.go

```diff
@@ -0,0 +1,584 @@
+			GinkgoT().Log(err)
+			return false
+		}
+		GinkgoT().Log("++++++++++", deviceLink.GetDeviceConnectedStatus())
+		return deviceLink.GetDeviceConnectedStatus() == metav1.ConditionTrue
+	}, 300, 1).Should(BeTrue())
+}
+func isDeviceConnectedFalse() {
+	var targetKey = types.NamespacedName{
+		Namespace: deviceLink.Namespace,
+		Name:      deviceLink.Name,
+	}
+	Eventually(func() bool {
+		if err := k8sCli.Get(testCtx, targetKey, &deviceLink); err != nil {
+			GinkgoT().Log(err)
+			return false
+		}
+		GinkgoT().Log("++++++++++", deviceLink.GetDeviceConnectedStatus())
+		return deviceLink.GetDeviceConnectedStatus() == metav1.ConditionFalse
+	}, 300, 1).Should(BeTrue())
+}
+
+func editGearToFast() {
+	var deviceLinkKey = types.NamespacedName{
+		Namespace: deviceLinkNamespace,
+		Name:      deviceLink.Name,
+	}
+	if err := k8sCli.Get(testCtx, deviceLinkKey, &deviceLink); err != nil {
+		GinkgoT().Log(err)
+		return
+	}
+	patch := []byte(`{"spec":{"template":{"spec":{"gear":"fast"}}}}`)
+	Expect(k8sCli.Patch(testCtx, &deviceLink, client.RawPatch(types.MergePatchType, patch))).Should(Succeed())
+}
+
+func isDeviceFast() {
+	var deviceList apiv1alpha1.DummySpecialDeviceList
+	Eventually(func() bool {
+		if err := k8sCli.List(testCtx, &deviceList, client.InNamespace("default")); err != nil {
+			GinkgoT().Log(err)
+			return false
+		}
+		for _, device := range deviceList.Items {
+			GinkgoT().Log("+++++Gear", device.Spec.Gear)
+			return device.Spec.Gear == "fast"
+		}
+		return false
+	}, 300, 1).Should(BeTrue())
+}
+
+func deleteDummyAdaptorPods() {
+	Expect(k8sCli.DeleteAllOf(testCtx, &corev1.Pod{}, client.InNamespace("octopus-system"), client.MatchingLabels{"app.kubernetes.io/name": "octopus-adaptor-dummy"})).
+		Should(Succeed())
+}
+
+func deleteLimbsPods() {
+	Expect(k8sCli.DeleteAllOf(testCtx, &corev1.Pod{}, client.InNamespace("octopus-system"), client.MatchingLabels{"app.kubernetes.io/component": "limb"})).
+		Should(Succeed())
+}
+
+func isDummyAdaptorPodsError() {
+	var podList corev1.PodList
+	Eventually(func() bool {
+		if err := k8sCli.List(testCtx, &podList, client.InNamespace("octopus-system")); err != nil {
+			GinkgoT().Log()
+			return false
+		}
+		for _, pod := range podList.Items {
+			if len(pod.Name) > 29 && pod.Name[:29] == "octopus-adaptor-dummy-adaptor" {
+				return pod.Status.Reason == "Error"
+			}
+		}
+		return false
+	}, 300, 1).Should(BeTrue())
+}
+
+func deleteNode() {
+	var anode = corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+		},
+	}
+	Expect(k8sCli.Delete(testCtx, &anode)).Should(Succeed())
+}
+
+func deleteDummyCrd() {
+	var crd = v1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "dummyspecialdevices.devices.edge.cattle.io",
+		},
+	}
+	Expect(k8sCli.Delete(testCtx, &crd)).Should(Succeed())
+}
+
+func redeployCrd() {
+	Expect(exec.RunKubectl(nil, GinkgoWriter, "apply", "-f", filepath.Join(testRootDir, "adaptors", "dummy", "deploy", "e2e", "all_in_one.yaml"))).
```

Comment:
There is a `testCurrDir` var, use it. At the same time, this is `all_in_one.yaml` file, not the CRD YAML file, please use the correct location.

Reply:
updated